pallets/alt-vr/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+        #[transactional]
+        pub fn update_vchatname(
+            origin: OriginFor<T>,
+            vchatname: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = vchatname.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = vchatname.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongVchatname);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.vchatname = vchatname.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrVchatnameUpdated(owner, vchatname);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Altvr was created. \[owner, username, vchatname\]
+        AltvrCreated(T::AccountId, ternoa::String, ternoa::String),
+        /// Altvr username updated  \[owner, username\]
+        AltvrUsernameUpdated(T::AccountId, ternoa::String),
+        /// Altvr vchatname updated  \[owner, vchatname\]
+        AltvrVchatnameUpdated(T::AccountId, ternoa::String),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Altvr username is too short.
+        TooShortUsername,
+        /// Altvr username is too long.
+        TooLongUsername,
+        /// Vchat name is too short.
+        TooShortVchatname,
+        /// Vchat name is too long.
+        TooLongVchatname,
+        /// User not found
+        UserNotFound,
+    }
+
+    /// List of Altvr datas create.
+    #[pallet::storage]
+    #[pallet::getter(fn altvrs)]
+    pub type Altvrs<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, AltvrData, OptionQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub altvrs: Vec<(T::AccountId, ternoa::String, ternoa::String)>,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                altvrs: Default::default(),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        fn build(&self) {
+            self.altvrs
+                .clone()
+                .into_iter()
+                .for_each(|(owner, username, vchatname)| {
+                    Altvrs::<T>::insert(owner.clone(), AltvrData::new(username, vchatname));
+                });
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    fn new_altvr(owner: &T::AccountId, username: &ternoa::String, vchatname: &ternoa::String) {
```

Comment:
If this is used just in once place, we can remove this function

Reply:
done